yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtTree.kt

```diff
@@ -597,22 +599,36 @@ internal class CrdtTreeNode(
     /**
      * Copies itself deeply.
      */
-    fun deepcopy(): CrdtTreeNode {
-        val clone = CrdtTreeNode(pos, type, _attributes = _attributes.deepCopy())
-        val node = this
-        return clone.apply clone@{
-            removedAt = node.removedAt
-            if (node.isText) {
-                value = node.value
+    fun deepCopy(): CrdtTreeNode {
+        return copy(
+            _value = _value,
+            childNodes = _children.map { child ->
+                child.deepCopy()
+            }.toMutableList(),
+            _attributes = _attributes.deepCopy(),
+        ).also {
+            it.size = size
+            it.removedAt = removedAt
+            it._children.forEach { child ->
+                child.parent = it
             }
-            size = node.size
-            node._children.map { child ->
-                child.deepcopy().apply {
-                    parent = this@clone
-                }
-            }.let(_children::addAll)
         }
     }
+
+    @Suppress("FunctionName")
+    companion object {
+
+        fun CrdtTreeText(pos: CrdtTreePos, value: String? = null): CrdtTreeNode {
+            return CrdtTreeNode(pos, DEFAULT_TEXT_TYPE, value)
+        }
+
+        fun CrdtTreeElement(
+            pos: CrdtTreePos,
+            type: String,
+            children: List<CrdtTreeNode> = emptyList(),
+            attributes: Rht = Rht(),
+        ) = CrdtTreeNode(pos, type, null, children.toMutableList(), attributes)
+    }
```

Comment:
@7hong13 I added helpers for CrdtTreeNode constructor as requirements for different type of nodes were not explicit.

Reply:
it looks much better👍